### PR TITLE
[TC-273] FE changes

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0273](https://bcdevex.atlassian.net/browse/TC-0273)

Objective:

- Edit profile pop up form 
- field validation
*note: we may want to look at the field validation/dto's for creating/updating personnel a bit more closely. ie: shoudl we disable the email edit field if this is going to be a unique field in order to prevent duplicate form submissions from CHEFS

### CODE REVIEW

File: frontend/src/pages/profile/ProfileEditPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    );
+
+    Object.keys(personnel).forEach((key) => {
+      if (values[key as keyof Personnel] === personnel[key as keyof Personnel]) {
+        delete values[key as keyof Personnel];
+      }
+    });
+
+    try {
+      const res = await AxiosPrivate.patch(
```
Comment (fw-noel) [inline frontend/src/pages/profile/ProfileEditPopup.tsx:59]: I'd suggest adding this to `usePersonnel`, to keep the requests in a place.

File: frontend/src/pages/profile/ProfileEditPopup.tsx [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      default:
+        return false;
+    }
+  };
+  const handleSubmit = async (
+    values: Personnel,
+    helpers: FormikHelpers<Personnel>,
+  ) => {
+    // only send the fields that have been changed
+    values.remoteOnly = stringToBoolean(values.remoteOnly ? 'true' : 'false');
```
Comment (fw-noel) [inline frontend/src/pages/profile/ProfileEditPopup.tsx:47]: These string to booleans feel like they should be unnecessary.. Why are they necessary?

**Review by fw-noel — COMMENTED** [review]
Some small comments.
Also, I noticed that when I click "Edit", the dialog starts a little bit scrolled down. 

Screenshot 2024-02-29 at 1 02 37 AM

Also, hitting "Edit" is minimizing the accordion. Nothing to do with fat fingering it. You'll want to use `stopPropagation` on the button, but watch that it also triggers `handleOpenEditPopUp` when you close the pop up so you might see some weird errors.

**Comment by chelsea-EYDS** [comment]
> Some small comments. Also, I noticed that when I click "Edit", the dialog starts a little bit scrolled down.
> 
> Screenshot 2024-02-29 at 1 02 37 AM
> Also, hitting "Edit" is minimizing the accordion. Nothing to do with fat fingering it. You'll want to use `stopPropagation` on the button, but watch that it also triggers `handleOpenEditPopUp` when you close the pop up so you might see some weird errors.

ugh, I really don't like having the edit button inside of the button that opens the accordian. It's generating warnings in the console re: invalid nesting of buttons as well.

File: frontend/src/pages/profile/ProfileEditPopup.tsx (hunk 2 above)
Comment (chelsea-EYDS) [reply frontend/src/pages/profile/ProfileEditPopup.tsx]: I'm using a Field from formik, the type for a value is "string|boolean|number". Even though the value is a boolean, when it's passed into the submit function it is a string, so I'm converting it to a boolean.

**Review by fw-noel — APPROVED** [review]

**Pull request merged**